std/assembly/internal/typedarray.ts

```diff
@@ -244,3 +244,14 @@ export function FOREACH<TArray extends TypedArray<T>, T>(
     callbackfn(LOAD<T>(buffer, i, byteOffset), i, array);
   }
 }
+
+@inline
+export function REVERSE<TArray extends TypedArray<T>, T>(array: TArray): TArray {
+  var buffer = array.buffer;
+  for (let front = 0, back = array.length - 1; front < back; ++front, --back) {
+    let temp = LOAD<T>(buffer, front);
+    STORE<T>(buffer, front, LOAD<T>(buffer, back));
+    STORE<T>(buffer, back, temp);
```

Comment:
`array.byteOffset`?

Reply:
I did forget about that! Whoops.

Reply:
This should be resolved. I added the `byteOffset` parameter to each function call.